web-services-app/src/test/java/edu/unc/lib/boxc/web/services/processing/SingleUseKeyServiceTest.java

```diff
@@ -0,0 +1,159 @@
+    }
+
+    @Test
+    public void testGenerateMultipleCallsForSameId() throws IOException {
+        var key1 = singleUseKeyService.generate(UUID_1);
+        var key2 = singleUseKeyService.generate(UUID_1);
+        var key3 = singleUseKeyService.generate(UUID_1);
+
+        var newRecords = parseCsv(CSV_HEADERS, csvPath);
+        var currentMilliseconds = System.currentTimeMillis();
+        assertCsvContainsCorrectEntry(newRecords, UUID_1, key1, currentMilliseconds);
+        assertCsvContainsCorrectEntry(newRecords, UUID_1, key2, currentMilliseconds);
+        assertCsvContainsCorrectEntry(newRecords, UUID_1, key3, currentMilliseconds);
+
+    }
+
+    @Test
+    public void testKeyIsValid() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        var futureExpiration = System.currentTimeMillis() + DAY_MILLISECONDS;
+        generateDefaultCsv(key, futureExpiration);
+        assertTrue(singleUseKeyService.keyIsValid(key));
+    }
+
+    @Test
+    public void testKeyIsNotValid() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        var timestamp = System.currentTimeMillis();
+        generateDefaultCsv(null, timestamp);
+        assertFalse(singleUseKeyService.keyIsValid(key));
+    }
+
+    @Test
+    public void testKeyIsNotValidCurrentTimeIsMoreThan24hLater() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        var pastTimestamp = System.currentTimeMillis() - (2 * DAY_MILLISECONDS);
+        generateDefaultCsv(key, pastTimestamp);
+        assertFalse(singleUseKeyService.keyIsValid(key));
+    }
+
+    @Test
+    public void testInvalidate() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        var expirationTimestamp = System.currentTimeMillis() + DAY_MILLISECONDS;
+        generateDefaultCsv(key, expirationTimestamp);
+        singleUseKeyService.invalidate(key);
+
+        var records = parseCsv(CSV_HEADERS, csvPath);
+        assertDoesNotContainValue(records, SingleUseKeyService.ACCESS_KEY, key);
+    }
+
+    @Test
+    public void testInvalidateWhenKeyIsNotPresent() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        var expirationTimestamp = System.currentTimeMillis()+ DAY_MILLISECONDS;
+        generateDefaultCsv(null, expirationTimestamp);
+        singleUseKeyService.invalidate(key);
+
+        var records = parseCsv(CSV_HEADERS, csvPath);
+        assertDoesNotContainValue(records, SingleUseKeyService.ACCESS_KEY, key);
+    }
+
+    private void assertDoesNotContainValue(List<CSVRecord> csvRecords, String column, String value) {
+        for (CSVRecord record : csvRecords) {
+            if (value.equals(record.get(column))) {
+                fail("Entry found!");
+                return;
+            }
+        }
+    }
+
+    private void assertCsvContainsCorrectEntry(List<CSVRecord> csvRecords,
+                                               String id, String key, long currentTimestamp) {
```

Comment:
I think it'd be okay to drop `currentTimestamp` from this method signature, since it's always being passed `System.currentTimeMillis()`, and the expiration delay is much longer than the amount of time that would pass between each assertion.